[FLINK-28511] Improve Spark dependencies and document (#210)
apache/flink-table-store: docs/content/docs/engines/spark.md

You are using an unreleased version of Table Store, you need to manually [Build Spark Bundled Jar]({{< ref "docs/engines/build" >}}) from the source code.
{{< /unstable >}}

Use `--jars` in spark-sql:
```bash
spark-sql ... --jars flink-table-store-spark-{{< version >}}.jar
```

> You can also copy `flink-table-store-spark-{{< version >}}.jar` to `spark/jars` in your Spark installation.

Reviewer: ```suggestion
Alternatively, you can copy `flink-table-store-spark-{{< version >}}.jar` under `spark/jars` in your Spark installation.
```

## Catalog

The following command registers the Table Store's Spark catalog with the name `table_store`:

```bash
spark-sql ... \
    --conf spark.sql.catalog.table_store=org.apache.flink.table.store.spark.SparkCatalog \
    --conf spark.sql.catalog.table_store.warehouse=file:/tmp/warehouse
```

> If you are using the Hive Metastore, you will need to add some configuration:

Reviewer: ```suggestion
Some extra configurations are needed if your Spark application uses the Hive Metastore to manage metadata.
```
